### ISSUE TITLE
Download to Home Assistant don't work

### ISSUE THREAD

**Issue by bjk201** [lavinir/hassio-onedrive-backup#185]
Hello,

I try to switch from Pi4 to a Proxmox VM. After generate a Fullback on PI and upload to one drive I receive the follow error on try download the File on Proxmox VM.

```
02/25/2024 22:17:24 [25] Info: Downloading backup hass_backup_2024-02-25-19-11.tar
02/25/2024 22:17:24 [17] Error: Error fetching backup hass_backup_2024-02-25-19-11.tar from Onedrive to Home Assistant. System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)
   at System.Uri..ctor(String uriString)
   at Microsoft.Kiota.Abstractions.RequestInformation.get_URI()
   at Microsoft.Kiota.Abstractions.Authentication.BaseBearerTokenAuthenticationProvider.AuthenticateRequestAsync(RequestInformation request, Dictionary`2 additionalAuthenticationContext, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.GetHttpResponseMessage(RequestInformation requestInfo, CancellationToken cancellationToken, Activity activityForAttributes, String claims, Boolean isStreamResponse)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendPrimitiveAsync[ModelType](RequestInformation requestInfo, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Graph.Drives.Item.Special.Item.Content.ContentRequestBuilder.GetAsync(Action`1 requestConfiguration, CancellationToken cancellationToken)
   at hassio_onedrive_backup.Graph.GraphHelper.DownloadFileAsync(String fileName, Action`1 progressCallback) in /src/Graph/GraphHelper.cs:line 264
   at hassio_onedrive_backup.Hass.BackupManager.DownloadBackupFromOneDrive(OnedriveBackup onlineBackup, Action`1 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 403
```

**Comment by lavinir** [comment]
You're right, this was introduced in the latest update with the migration to the new libraries.

**Comment by bjk201** [comment]
All System are up2date. 

Old Pi4 System:

Core
2024.2.3
Supervisor
2024.02.0
Operating System
11.5
Frontend
20240207.1

Your Addon: 2.3


The new Proxmox System Running on:

Core
2024.2.3
Supervisor
2024.02.0
Operating System
11.5
Frontend
20240207.1

Your Addon: 2.3


-----------


My Backup File size are 17gb.

**Comment by lavinir** [comment]
Yes if it was unclear in my response, I meant the bug was introduced in the latest update. 
I will push a fix for this in the next update.

**Comment by bjk201** [comment]
Thank you :)

**Comment by lavinir** [comment]
fixed in 2.3.1
